flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/AdaptiveBatchSchedulerFactory.java

```diff
@@ -162,6 +147,74 @@ public SchedulerNG createInstance(
+                log,
+                jobGraph,
+                ioExecutor,
+                jobMasterConfiguration,
+                futureExecutor,
+                userCodeLoader,
+                checkpointRecoveryFactory,
+                rpcTimeout,
+                blobWriter,
+                jobManagerJobMetricGroup,
+                shuffleMaster,
+                partitionTracker,
+                executionDeploymentTracker,
+                initializationTimestamp,
+                mainThreadExecutor,
+                jobStatusListener,
+                failureEnrichers,
+                blocklistOperations,
+                new DefaultExecutionOperations(),
+                allocatorFactory,
+                restartBackoffTimeStrategy,
+                new ScheduledExecutorServiceAdapter(futureExecutor),
+                DefaultVertexParallelismAndInputInfosDecider.from(
+                        getDefaultMaxParallelism(jobMasterConfiguration, executionConfig),
+                        jobMasterConfiguration));
+    }
+
+    public AdaptiveBatchScheduler createScheduler(
+            Logger log,
+            JobGraph jobGraph,
+            Executor ioExecutor,
+            Configuration jobMasterConfiguration,
+            ScheduledExecutorService futureExecutor,
+            ClassLoader userCodeLoader,
+            CheckpointRecoveryFactory checkpointRecoveryFactory,
+            Time rpcTimeout,
+            BlobWriter blobWriter,
+            JobManagerJobMetricGroup jobManagerJobMetricGroup,
+            ShuffleMaster<?> shuffleMaster,
+            JobMasterPartitionTracker partitionTracker,
+            ExecutionDeploymentTracker executionDeploymentTracker,
+            long initializationTimestamp,
+            ComponentMainThreadExecutor mainThreadExecutor,
+            JobStatusListener jobStatusListener,
+            Collection<FailureEnricher> failureEnrichers,
+            BlocklistOperations blocklistOperations,
+            ExecutionOperations executionOperations,
+            ExecutionSlotAllocatorFactory allocatorFactory,
+            RestartBackoffTimeStrategy restartBackoffTimeStrategy,
+            ScheduledExecutor delayExecutor,
+            VertexParallelismAndInputInfosDecider vertexParallelismAndInputInfosDecider)
+            throws Exception {
+
+        checkState(
+                jobGraph.getJobType() == JobType.BATCH,
+                "Adaptive batch scheduler only supports batch jobs");
+        checkAllExchangesAreSupported(jobGraph);
+
+        final boolean enableSpeculativeExecution =
+                jobMasterConfiguration.get(BatchExecutionOptions.SPECULATIVE_ENABLED);
+
+        final HybridPartitionDataConsumeConstraint hybridPartitionDataConsumeConstraint =
+                getOrDecideHybridPartitionDataConsumeConstraint(
+                        jobMasterConfiguration, enableSpeculativeExecution);
+
+        final Consumer<ComponentMainThreadExecutor> combinedStartUpActions = m -> {};
```

Comment:
It's no longer a **combined**StartUpActions. Maybe directly pass `m -> {}` as the ctor param?